packages/runtime/src/next/middleware/index.ts

```diff
@@ -0,0 +1,36 @@
+import { NextResponse, type NextMiddleware } from 'next/server'
+import { withDraftMiddleware } from './draft-middleware'
+import { withBrandingMiddleware } from './branding-middleware'
+
+type MiddlewareDecorator = (middleware: NextMiddleware) => NextMiddleware
+
+function composeMiddlewares(
+  baseMiddleware: NextMiddleware,
+  firstMiddlewareDecorator: MiddlewareDecorator,
+  ...otherMiddlewareWrappers: MiddlewareDecorator[]
+) {
+  const middlewares = otherMiddlewareWrappers.reduce(
+    (accumulatedMiddlewares, nextMiddleware) => middleware =>
+      accumulatedMiddlewares(nextMiddleware(middleware)),
+    firstMiddlewareDecorator,
+  )
+
+  return middlewares(baseMiddleware)
+}
+
+const passThroughMiddleware: NextMiddleware = (_req, _event) => NextResponse.next()
```

Comment:
Doesn't this have to pass `request` to `NextResponse.next` for it to be a real passthrough?